src/services/ExposureNotificationService/ExposureNotificationService.ts

```diff
@@ -239,6 +239,54 @@ export class ExposureNotificationService {
     await this.recordKeySubmission();
   }
 
+  calculateNeedsSubmission(exposureStatus: ExposureStatus, today: Date): boolean {
+    if (exposureStatus.type !== ExposureStatusType.Diagnosed) return false;
```

Comment:
Just noting the change here we used to have this inline.  Unclear if given the Observable the value could possibly be different vs the parameter (would need some testing).

```
const exposureStatus = this.exposureStatus.get();
```

Reply:
Old

``` 
if (currentStatus.type === ExposureStatusType.Diagnosed) {
      const today = getCurrentDate();
      const cycleEndsAt = new Date(currentStatus.cycleEndsAt);
      // There is a case where using UTC and device timezone could mess up user experience. See `date-fn.spec.ts`
      // Let's use device timezone for resetting exposureStatus for now
      // Ref https://github.com/cds-snc/covid-shield-mobile/issues/676
      if (daysBetween(today, cycleEndsAt) <= 0) {
        this.exposureStatus.set({type: ExposureStatusType.Monitoring, lastChecked: currentStatus.lastChecked});
        return this.finalize();
      }
```

New including Master - looks like this would reach detectExposure when Diagnosed vs the old code wouldn't


``
 const updatedExposure = this.updateExposure(currentExposureStatus, today);
    if (updatedExposure !== currentExposureStatus) {
      this.exposureStatus.set(updatedExposure);
    }

    const {keysFileUrls, lastCheckedPeriod} = await this.getKeysFileUrls();

    try {
      captureMessage('lastCheckedPeriod', {lastCheckedPeriod});
      const summaries = await this.exposureNotification.detectExposure(exposureConfiguration, keysFileUrls);
```